.travis.yml

```diff
@@ -58,15 +40,34 @@ jobs:
       cache: npm
       before_install:
         - cd frontend/
-
       install:
         - npm install
-
       script:
         - npm run test:unit
-        - npm run test:e2e:ci
-
       after_script:
         - cat ./data/coverage/lcov.info | coveralls ..
 
+    - name: "E2E tests"
+      stage: test
+      install:
+        - docker-compose up -d --build composer backend frontend db
+        - bash wait-for-container-startup.sh
+      before_script:
+        - cd frontend/
+      script:
+        - docker run -v $PWD:/e2e -w /e2e --network host -e CYPRESS_BASE_URL=http://localhost:3000 cypress/included:3.4.0
```

Comment:
Habe jetzt noch nicht wirklich verstanden, was da abgeht. Aber scheint zu funktionieren.

Reply:
Du hattest zuvor `npm run test:e2e:ci` drin. Das startet aber einen neuen Webpack Dev Server, was nur für Frontend-Apps ohne Backend sinnvoll ist.
Ich habs auf die schnelle nicht hingekriegt das installierte Cypress aus dem Frontend-Container von Travis aus zu verwenden, darum lasse ich hier Cypress in einem weiteren Container laufen, der spezifisch dafür gemacht ist: `docker run cypress/included:3.4.0`
`-v $PWD:/e2e -w /e2e` macht das aktuelle Verzeichnis im Container verfügbar, das ist nötig damit Cypress cypress.json und die Tests lesen kann die es ausführen soll.
Mittels `--network host -e CYPRESS_BASE_URL=...` teile ich Cypress mit, wo die Applikation ist die es testen soll.